kola/tests/misc/nvidia.go

```diff
@@ -3,50 +3,239 @@ package misc
 import (
 	"bytes"
 	"fmt"
+	"strings"
 	"time"
 
+	"github.com/coreos/go-semver/semver"
 	"github.com/coreos/pkg/capnslog"
 	"github.com/flatcar/mantle/kola"
 	"github.com/flatcar/mantle/kola/cluster"
 	"github.com/flatcar/mantle/kola/register"
+	testsutil "github.com/flatcar/mantle/kola/tests/util"
+	"github.com/flatcar/mantle/platform"
+	"github.com/flatcar/mantle/platform/conf"
 	"github.com/flatcar/mantle/util"
 )
 
 const (
 	CmdTimeout = time.Second * 300
 )
 
+const nvidiaDriverVersionOverride = `
+variant: flatcar
+version: 1.0.0
+storage:
+  files:
+  - path: /etc/flatcar/nvidia-metadata
+    contents:
+      inline: |
+        {{ .NVIDIA_DRIVER_VERSION_LINE }}
+`
+
+const nvidiaOperatorTemplate = `
+variant: flatcar
+version: 1.0.0
+
+storage:
+  files:
+  - path: /etc/flatcar/nvidia-metadata
+    contents:
+      inline: |
+        NVIDIA_DRIVER_VERSION=570.86.15
+  - path: /opt/extensions/kubernetes-v1.30.4-{{ .ARCH_SUFFIX }}.raw
```

Comment:
Is there a reason for picking exactly this version? As opposed to, say, 1.30.8 or any newer major version. Asking, because I think that eventually we will need to be bumping those versions, so maybe in future we will want to change the code to always download the latest version of k8s or nvidia runtime.

Reply:
any version is good enough, i've bumped it to 1.30.8.

I think we'll need to update this eventually but i don't expect it to have to always be the latest - the GPU operator itself has a matrix of supported k8s version.

I think this test is going to be most helpful to test the nvidia-runtime sysext and any changes to flatcar's nvidia.service that may break the GPU operator.